src/cpu/pred/BranchPredictor.py

```
@@ -1060,16 +1060,33 @@ class BTBTAGE(TimedBaseBTBPredictor):
-    TTagPcShifts = [1]
+class MicroTAGE(TimedBaseBTBPredictor):
+    """Micro-sized BTB TAGE predictor used alongside uBTB"""
+    type = 'MicroTAGE'
+    cxx_class = 'gem5::branch_prediction::btb_pred::MicroTAGE'
+    cxx_header = "cpu/pred/btb/microtage.hh"
+
+    needMoreHistories = Param.Bool(True, "MicroTAGE needs more histories")
+    enableSC = Param.Bool(False, "Enable SC or not")
+    updateOnRead = Param.Bool(True,"Enable update on read, no need to save tage meta in FTQ")
+    # Keep vector parameters consistent with numPredictors to avoid constructor asserts.
+    numPredictors = Param.Unsigned(4, "Number of TAGE predictors")
```

Ruff (0.15.5)

[error] `Param` may be undefined, or defined from star imports

(F405)